exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -134,20 +134,68 @@ func (prwe *PRWExporter) PushMetrics(ctx context.Context, md pdata.Metrics) erro
 
 					// handle individual metric based on type
 					switch metric.DataType() {
-					case pdata.MetricDataTypeDoubleSum, pdata.MetricDataTypeIntSum, pdata.MetricDataTypeDoubleGauge, pdata.MetricDataTypeIntGauge:
-						if err := prwe.handleScalarMetric(tsMap, resource, metric); err != nil {
+					case pdata.MetricDataTypeDoubleGauge:
+						dataPoints := metric.DoubleGauge().DataPoints()
+						if dataPoints.Len() == 0 {
 							dropped++
-							errs = append(errs, consumererror.Permanent(err))
+							errs = append(errs, consumererror.Permanent(fmt.Errorf("empty data points. %s is dropped", metric.Name())))
 						}
-					case pdata.MetricDataTypeHistogram, pdata.MetricDataTypeIntHistogram:
-						if err := prwe.handleHistogramMetric(tsMap, resource, metric); err != nil {
+						for x := 0; x < dataPoints.Len(); x++ {
+							addSingleDoubleDataPoint(dataPoints.At(x), resource, metric, prwe.namespace, tsMap, prwe.externalLabels)
+						}
+					case pdata.MetricDataTypeIntGauge:
+						dataPoints := metric.IntGauge().DataPoints()
+						if dataPoints.Len() == 0 {
+							dropped++
+							errs = append(errs, consumererror.Permanent(fmt.Errorf("empty data points. %s is dropped", metric.Name())))
+						}
+						for x := 0; x < dataPoints.Len(); x++ {
+							addSingleIntDataPoint(dataPoints.At(x), resource, metric, prwe.namespace, tsMap, prwe.externalLabels)
+						}
```

Comment:
I think we can reduce some of the duplicate code here by having a func that accepts IntDataPointSlice. So this and IntSum will call into the same func, as well as for double.

```
func addIntDataPoints(dataPoints pdata.IntDataPointSlice, ...) error {
	if dataPoints.Len() == 0 {
		dropped++							
		return consumererror.Permanent(fmt.Errorf("empty data points. %s is dropped", metric.Name()))
	}
	for x := 0; x &lt; dataPoints.Len(); x++ {	
		addSingleIntDataPoint(dataPoints.At(x), resource, metric, prwe.namespace, tsMap, prwe.externalLabels)
	}
	return nil
}
```

Reply:
Actually `addSingleIntDataPoint` requires the metric as a parameter. So I both passed datapoints and metric to the new function.

Reply:
Updated in https://github.com/open-telemetry/opentelemetry-collector/pull/3261/commits/dcbc7ffc21dfc85e58b9ecdf4485106579a6d675